File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/ipAllowlist/TenantIpAllowlistRepository.java
package br.com.deltaglobalbank.identity.domain.ipAllowlist;

import java.util.List;
import java.util.UUID;

public interface TenantIpAllowlistRepository {
    TenantIpAllowlist save(TenantIpAllowlist entry);

    TenantIpAllowlist findById(UUID id);

    List<TenantIpAllowlist> findAllByTenantId(UUID tenantId);

    boolean existsByCidrAndTenantId(String cidr, UUID tenantId);

    void delete(UUID id);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/ipAllowlist/TenantIpAllowlistRepository.kt
package br.com.deltaglobalbank.identity.domain.ipAllowlist

import java.util.UUID

interface TenantIpAllowlistRepository {
    fun save(entry: TenantIpAllowlist): TenantIpAllowlist
    fun findById(id: UUID): TenantIpAllowlist?
    fun findAllByTenantId(tenantId: UUID): List<TenantIpAllowlist>
    fun existsByCidrAndTenantId(cidr: String, tenantId: UUID): Boolean
    fun delete(id: UUID)
}